File: 3163-string-compression-iii/3163-string-compression-iii.cpp
class Solution {
public:
    string compressedString(string word) {
        string comp = "";
        int len = 1;
        
        for(int i=0;i<word.size()-1;i++){
            if(len<9 && word[i] == word[i+1]){
                len++;
            }
            else{
                comp += to_string(len) + word[i];
                len = 1;
            }
        }
        comp += to_string(len) + word[word.size()-1];
        return comp;
    }
};

File: 3163-string-compression-iii/3163-string-compression-iii.java
class Solution {
    public String compressedString(String word) {
        StringBuilder ans = new StringBuilder();
        int n = word.length();
        int len = 1;
        
        for(int i=0;i<n-1;i++){
            if(len<9 && word.charAt(i) == word.charAt(i+1)){
                len++;
            }
            else{
                ans.append(len);
                ans.append(word.charAt(i));
                len = 1;
            }
        }
        ans.append(len);
        ans.append(word.charAt(n-1));
        return ans.toString();
    }
}